pkg/config/template.go

```diff
@@ -225,28 +198,8 @@ func GetTemplateCMName(name string) string {
 	return TemplatePrefix + name
 }
 
-func GetTempCMName(templateName, filename string, templateData []byte) string {
-	return fmt.Sprintf("rehearse-%s-%s", inputHash([]byte(filename), templateData), templateName)
-}
-
-// oneWayEncoding can be used to encode hex to a 62-character set (0 and 1 are duplicates) for use in
-// short display names that are safe for use in kubernetes as resource names.
-var oneWayNameEncoding = base32.NewEncoding("bcdfghijklmnpqrstvwxyz0123456789").WithPadding(base32.NoPadding)
-
-// inputHash returns a string that hashes the unique parts of the input to avoid collisions.
-func inputHash(inputs ...[]byte) string {
-	hash := sha256.New()
-
-	// the inputs form a part of the hash
-	for _, s := range inputs {
-		hash.Write(s)
-	}
-
-	// Object names can't be too long so we truncate
-	// the hash. This increases chances of collision
-	// but we can tolerate it as our input space is
-	// tiny.
-	return oneWayNameEncoding.EncodeToString(hash.Sum(nil)[:5])
+func GetTempCMName(templateName, hash string) string {
+	return fmt.Sprintf("rehearse-%s-%s", hash[:8], templateName)
```

Comment:
Can this segfault?

Reply:
First, I don't like this as a free function with seemingly disjoint parameters, which is why in the following PR it becomes a struct method: https://github.com/openshift/ci-operator-prowgen/pull/168/commits/3bc34181ab976d3c4b9c82a4fc62b1f29a8b74e9#diff-33dd159268780d36aa2b7bd9253f7a29L42.

We should not have that problem with the data flow we have. `hash` comes directly from the `git` command output, which has a well-defined format. We will either not have a template or have one with file name and hash. Someone could still call this with an inappropriate value (or an initialized struct, in #168), but a segfault is the right outcome in that situation IMO.